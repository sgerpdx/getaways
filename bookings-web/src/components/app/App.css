body {
  background-color: grey;
  display: flex;
  flex-wrap: wrap;
}

.listContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
}

.placeItemList {
  background-color: darkolivegreen;
  color: white;
  display: flex;
  flex-direction: column;
  width: 360px;
  height: 360px;
  margin: 4px;
  padding: 4px;
  list-style-type: none;
}

/* .placeHeading {
  list-style-type: none;
} */

.placeDescription {
  border: solid 1px rgb(198, 205, 212);
  border-radius: 4px;
  padding: 4px;
}
