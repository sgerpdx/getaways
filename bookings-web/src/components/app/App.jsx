import React, { useState, useEffect } from 'react';
import Getaways from '../../containers/Getaways';
import DetailContainer from '../../containers/DetailContainer';
import { Route, BrowserRouter as Router } from 'react-router-dom';
import Header from '../../containers/Header';
import Login from '../login/Login';
import {
  getUserFromLocalStorage,
  putUserInLocalStorage,
} from '../../utils/LocalStorage-Utils';

export default function App() {
  const [user, setUser] = useState('');

  useEffect(() => {
    getUserFromLocalStorage();
    setUser;
  }, []);
  //do we want those empty brackets in this case?

  const handleUserChange = (user) => {
    setUser();
    putUserInLocalStorage();
  };

  const handleLogout = () => {
    localStorage.clear();
    handleUserChange();
  };

  //add a PrivateRoute once bookings become a thing:

  return (
    <Router>
      <h2>Getaways</h2>
      <Header user={user} handleLogout={handleLogout} />
      <Route exact path="/" component={Getaways} />
      <Route exact path="/:id" component={DetailContainer} />
      <Route
        path="/login"
        exact
        render={(routerProps) => (
          <Login handleUserChange={handleUserChange} {...routerProps} />
        )}
      />
      {/* <Route exact path="/login" component={Login} /> */}
    </Router>
  );
}
