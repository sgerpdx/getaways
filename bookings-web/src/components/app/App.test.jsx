import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import App from './App';
jest.useFakeTimers();

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    return waitFor(() => {
      const { asFragment } = render(<App />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
