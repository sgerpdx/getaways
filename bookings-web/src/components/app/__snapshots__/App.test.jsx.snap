// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <h2>
    Getaways
  </h2>
  <a
    aria-role="redirect to haunted house"
    href="/"
  >
    <h4>
      Home
    </h4>
  </a>
  <span>
    waiting example for test
  </span>
  <hr />
  <h2>
    Loading...
  </h2>
</DocumentFragment>
`;
